### PR TITLE
Feat/AUT-741/inject rich passage styles in itemData

### PR DESCRIPTION
**Pull request by bziondik**
Related to: https://oat-sa.atlassian.net/browse/AUT-741

Inject rich passage stylesheets as item stylesheets in ItemData with absolute URL

need branches:
https://github.com/oat-sa/extension-tao-itemqti/tree/feature/CON-57%2Frich-passage-authoring
https://github.com/oat-sa/extension-tao-mediamanager/tree/feature/CON-57%2Frich-passage-authoring

How to test:
1. create Rich passages with own styles (use Style editor)
2.  add these Rich passages in item
3. save item
4. refresh page
5. preview item, Rich passage styles should be applied

### CODE REVIEW

File: views/js/previewer/helpers/richPassages.js [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2021 (original work) Open Assessment Technologies SA;
+ */
+
+define(['lodash', 'uri', 'util/url', 'core/dataProvider/request'], function (_, uri, urlUtil, request) {
+    'use strict';
+
+    function getPassagesFromElement(element = {}) {
```
Comment (AnastasiaArcadia) [inline views/js/previewer/helpers/richPassages.js:22]: Could you please add jsdoc for this function as well?

File: views/js/test/previewer/helpers/richPassages/test.html [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        <script type="text/javascript" src="js/lib/require.js"></script>
+        <script type="text/javascript" src="js/lib/qunit/qunit-parameterize.js"></script>
+
+        <script  type="text/javascript">
+
+             //don't start the test now
+            QUnit.config.autostart = false;
+
+            //load the config
+            require(['/tao/ClientConfig/config'], function(){
```
Comment (AnastasiaArcadia) [inline views/js/test/previewer/helpers/richPassages/test.html:20]: Missed indents

File: views/js/previewer/helpers/richPassages.js [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+        let includes = {};
+        _.forEach(['elements', 'choices'], elementCollection => {
+            if (elementCollection === 'choices' && _.isArray(element[elementCollection])) {
+                // in MatchInterection choices is Array of match sets
+                _.forEach(element[elementCollection], choiceMatch => {
```
Comment (AnastasiaArcadia) [inline views/js/previewer/helpers/richPassages.js:27]: Can we move to a single style, either `for` or `forEach`?

**Review by AnastasiaArcadia — APPROVED** [review]
Works as expeceted, good job!

- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [x] Pull request's target is not master

**Review by Silvia-loza — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [X] New code is respecting code style rules
- [X] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [X] Feature is working correctly on my local machine (if applicable)
- [X] Acceptance criteria are respected
- [X] Pull request title and description are meaningful

**Pull request merged**